docs/docs/code/python/README.md

```diff
@@ -0,0 +1,279 @@
+
+```json
+{
+  "id": 1,
+  "name": "Bulbasaur",
+  "type": "plant"
+}
+```
+
+Every Python step can import the `pipedream.script_helpers` module. This module contains data from other steps in the workflow via `steps` dictionary.
+
+Here's an example of how to get the ID from the Pokemon we received in the trigger.
+
+```python
+from pipedream.script_helpers import (steps, export)
+
+# retrieve the data points from the HTTP request in the initial workflow trigger 
+name = steps["trigger"]["event"]["name"]
+pokemon_type = steps["trigger"]["event"]["type"]
+
+print(f"{pokemon_name} is a {pokemon_type} type Pokemon")
+```
+
+### Sending data downstream to other steps
+
+The trigger steps automatically export their data for other steps to use downstream.
+
+Just pass the data to the `export` module from `pipedream.script_helpers`.
+
+```python
+# this step is named "code" in the workflow
+from pipedream.script_helpers import (steps, export)
+
+pokemon = { name: "Pikachu" }
+
+export('pokemon', pokemon)
+```
+
+Now this `pokemon` is accessible to downstream steps within `steps.code.pokemon`
+
+::: warning
+Not all data types can be stored in the `steps` data shared between workflow steps.  Only JSON serializable types can be passed, which includes:
+
+* lists 
+* dictionaries
+  :::
+
+## Using environment variables
+
+You can leverage any [environment variables defined in your Pipedream account](/environment-variables/#environment-variables) in a python step. This is useful for keeping your secrets out of code as well as keeping them flexible to swap API keys without having to update each step individually.
+
+To access them, use the `os` module.
+
+```python
+import os
+import requests
+
+token = os.environ['TWITTER_API_KEY']
+
+url = 'https://api.twitter.com/2/users/@pipedream/mentions'
+
+headers { 'Authorization': f"Bearer {token}"}
+r = requests.get(url, headers=headers)
+
+print(r.text)
+```
+
+:::tip
+There are 2 different ways of using the `os` module to access your environment variables.
+
+`os.environ['ENV_NAME_HERE']` will raise an error that stops your workflow if that key doesn't exist in your Pipedream account.
+
+Whereas `os.environ.get('ENV_NAME_HERE')` will _not_ throw an error and instead returns an empty string. 
+
+If your code relies on the presence of a environment variable, consider using `os.environ['ENV_NAME_HERE']` instead.
+:::
+
+## Handling errors
+
+You may find a need to exit a workflow early. In a Python step, just a `raise` an error to halt a step's execution.
+
+
+```python
+raise NameError('Something happened that should not. Exiting early.')
+```
+
+All exceptions from your Python code will appear in the logs area of the results.
+
+## File storage
+
+Not only can you run Python code in workflows, but you can also store files as well. This means you can upload or download photos, retrieve datasets or accept files from an HTTP request and much more.
```

Comment:
"Not only can you run Python code in workflows, but you can also store files as well" -> "You can also store files in Python code steps"

Reply:
Thanks, maybe I should look into a grammar GH action plugin 😅 

Just one minor change I made to the suggestion: "You can also store files in Python code steps" -> "You can also store and read files with Python steps."

My thinking is, we're not literally storing the file in a code step we're leveraging Python to manage files.